### PR TITLE
feat: migrate store/collections to typescript 

### LINKED ISSUE #732: Convert to typescript - store/collections

### PR DESCRIPTION
**Pull request by zakariaelas**
fixes #732

### CODE REVIEW

File: packages/medusa/src/api/routes/store/collections/list-collections.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  const [collections, count] = await productCollectionService.listAndCount(
+    selector,
+    listConfig
+  )
+
+  res.status(200).json({ collections, count, limit, offset })
+}
+
+export class StoreGetCollectionsParams {
```
Comment (sebastiannicolajsen) [inline packages/medusa/src/api/routes/store/collections/list-collections.ts:49]: Nit / Suggestion; Could we create a class similar to the `FindConfig<Entity>` interface exported [here](https://github.com/medusajs/medusa/blob/typescript/packages/medusa/src/types/common.ts) for pagination in endpoints?

File: packages/medusa/src/api/routes/store/collections/list-collections.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  @IsOptional()
+  @Type(() => Number)
+  limit?: number = 10
```
Comment (sebastiannicolajsen) [inline packages/medusa/src/api/routes/store/collections/list-collections.ts:52]: Question; I do not know if necessary (but we do it other places): add `@IsInt()` to both of these?

**Review by sebastiannicolajsen — COMMENTED** [review]
Generally LGTM

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/medusa-js/src/resources/collections.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    if (query) {
+      const queryString = Object.entries(query).map(([key, value]) => {
+        return `${key}=${value}`;
+      });
+
+      path = `/store/collections?${queryString.join('&')}`;
+    }
+
+    return this.client.request('GET', path);
```
[the remark this answers is not in the document]
Comment (zakariaelas) [reply packages/medusa-js/src/resources/collections.ts:32]: definitely! I believe another PR will fix all linting/prettier issues with the medusa-js client right @olivermrbl ?

File: packages/medusa/src/api/routes/store/collections/list-collections.ts (hunk 2 above)
Comment (zakariaelas) [reply packages/medusa/src/api/routes/store/collections/list-collections.ts:56]: Thanks for pointing it out! Adding `@IsInt()` is pretty important as it helps prevent query params such as `?offset=bla&limit=blo` from going down the service layer :+1:

**Pull request merged**